### ISSUE TITLE
Repeater click command

### ISSUE THREAD

**Issue by OpticNectar** [UXDivers/Grial-UI-Kit-Support#220]
How do you handle click events/commands on the `Repeater` element? The example shows a click event on the template itself, but that requires all sorts of extra binding etc, and seems really inefficient, especially since other elements in the kit have the `ItemClickCommand`.

**Comment by LeoHere** [comment]
@OpticNectar agree we should add such a property. Unfortunately, it is no there. So you should use the gesture recognizer in the template and bind a `Command` to it.

**Comment by OpticNectar** [comment]
@LeoHere I can try and do that for now. The issue is it really makes a lot of work and makes it harder to reuse templates. For example i'm using the same template for `GridOptionsView` and `Repeater`, and for `GridOptionsView` i'm using `ItemClickCommand`, so i'd have to duplicate the template to add the click command there just for the `Repeater` control.

**Comment by BradleyDHobbs** [comment]
Do you have an example of this? Trying to add a click event for a cardView inside of a repeater

**Comment by uzzu96** [comment]
Yes please. Can we have an example of binding a Tapped command to a viewmodel from a cardview inside a repeater?

**Comment by dirivero** [comment]
There's a way to simulate that command from Xaml using a gesture recognizers and reusing the items:

```
<grial:Repeater
    x:Name="repeater"
    ....
    BindingContext="{ Binding . }">
    <grial:Repeater.ItemTemplate>
        <DataTemplate>
            <local:MyTemplate>
                <ContentView.GestureRecognizers>
                    <TapGestureRecognizer
                        Command="{ Binding BindingContext.ItemTapCommand, Source={x:Reference repeater} }"
                        CommandParameter="{ Binding . }"/>
                </ContentView.GestureRecognizers>
            </local:MyTemplate>
        </DataTemplate>
    </grial:Repeater.ItemTemplate>
</grial:Repeater>
```

This binds the item tap event to a command named `ItemTapCommand` from the binding context of the page (its view-model). It also sends the specific item object as parameter so you know what was tapped.

Also, the repeater provides another way to achieve this by using the `ItemSelectedCommand`:

```
<grial:Repeater
    ...
    ItemSelectedCommand="{ Binding ItemSelectedCommand }">
    <grial:Repeater.ItemTemplate>
        <DataTemplate>
            <local:MyTemplate />
        </DataTemplate>
    </grial:Repeater.ItemTemplate>
</grial:Repeater>
```

The benefit of this command is that it won't execute twice even if the user quickly taps multiple times as the item is selected just the first time. Make sure to clear the selection if/when you actually want to enable another execution on the same item. If that's your case, just define a `Mode=TwoWay` binding from the repeater's `SelectedItem` to a view-model property and set it to null from the view-model itself.

Hope this helps.

**Comment by ddtechboy** [comment]
`ItemSelectedCommand` doesn't work for me. However, the `GestureRecognizers` technique worked well. Thanks, @dirivero for that.

But, can someone please explain why the former doesn't work?

**Comment by dirivero** [comment]
Did you set `SelectionMode="Single"` in the repeater? By default selection is disabled. I forgot to mention that.
